### ISSUE TITLE
Feature Request: allow `git pull --ff-only`, or allow custom GIt actions

### ISSUE THREAD

**Issue by rmunn** [Microsoft/vscode#10008]
- VSCode Version: Version 1.3.1 (commit e6b4afa)
- OS Version: Ubuntu 14.04

When I work on a Git repo that other people are pushing to, I prefer to use `git pull --ff-only` as my primary pull command, because I want to have the option of merging or rebasing depending on what the situation is. So if a fast-forward merge is not possible, I don't want Git to automatically create a merge commit for me; I want to examine the situation and decide what to do for myself.

Therefore, neither VS Code's "Pull" or "Pull (Rebase)" commands are really suitable for my preferred workflow. I would like one of the following features:

1. (My preference): Allow me to specify a custom command to be added to VS Code's Git menu (the dropdown menu that you get by clicking on the three-dots icon). Bonus points if I can specify the position of the custom command (i.e., below Sync but above Pull). Then I could add `Pull (FF only)` to the menu and be able to use my preferred Git workflow.

2. Allow me to customize the options on the Git command that is run when I choose one of the menu items. E.g., allow me to customize a `git.pull.options` config to add `--ff-only`. Then the "Pull" command that's already in the Git menu would do what I want it to do -- pull only if a fast-forward is possible, otherwise fetch and allow me to fo a merge or rebase manually.

3. Add a "Pull (FF only)" command to the Git menu, which would run `git pull --ff-only`.

Currently I can do what I want by opening the integrated terminal and running `git pull --ff-only` on the command line, so this isn't a blocker for me. But it means I'm basically avoiding using VS Code's Git integration, because sometimes it will do the wrong thing.

**Comment by kieferrm** [comment]
@rmunn have you looked into using `git config` and setting `pull.ff` to `only`? More [here](https://git-scm.com/docs/git-config).

**Comment by rmunn** [comment]
I'm currently on a project that's targeting Ubuntu Trusty, so my development machine is still on Trusty and so the version of Git I have installed is 1.9. The `pull.ff` config option was introduced in Git 2.x (according to http://stackoverflow.com/a/35298178/2314532), so I don't have it available. However, that *is* the right answer for my workflow. Once my current project is released and I'm able to upgrade to a more recent Linux distro, I'll have Git 2.x available and I'll be able to use the `pull.ff` config option to do what I want.

I've also verified that using the "Sync" command, or clicking on the convenient sync icon in the lower left corner of VS Code, also does the right thing when I've set `pull.ff`: it will refuse to pull if a fast-forward is impossible, which aborts the sync operation.

So as far as I'm concerned, this issue can be closed. I'll leave it open in case you decide that a `git.pull.options` config would be a good thing to add to VS Code, or if you want to allow the user to specify custom Git commands to add to the VS Code menu, in which case you might want this issue open to track that. But if you decide not to add any of those, feel free to close this issue.
